### PR TITLE
Fenwick tree with linear build

### PR DESCRIPTION
**Pull request by cameroncuster**
Updated the Fenwick Tree to have linear build from array O(n) with O(1) extra space by storing the prefix sum in the BIT array.

Passing Test:
https://judge.yosupo.jp/submission/56412

Algorithmic Details:
https://codeforces.com/blog/entry/59305

### CODE REVIEW

File: rangeDS/fenwickTree.cpp [hunk 1]
```diff
@@ -1,6 +1,16 @@
 struct fenwickTree {
 	vector<ll> bit;
 	fenwickTree(int n) : bit(n,0) {}
+	fenwickTree(const vector<int> & a) : bit(a.size()) {
```
Comment (lrvideckis) [inline rangeDS/fenwickTree.cpp:4]: ll

File: rangeDS/fenwickTree.cpp [hunk 2]
```diff
@@ -1,6 +1,16 @@
...
+		bit[0] = a[0];
```
Comment (lrvideckis) [inline rangeDS/fenwickTree.cpp:5]: This will RTE if a is empty, let's not have this behavior

File: rangeDS/fenwickTree.cpp [hunk 3]
```diff
@@ -1,6 +1,16 @@
...
+		for (int i = 1; i < (int)a.size(); i++)
+			bit[i] = bit[i-1] + a[i];
+		for (int i = a.size() - 1; i > 0; i--) {
```
Comment (lrvideckis) [inline rangeDS/fenwickTree.cpp:8]: Let's cast a.size to an int, otherwise this will underflow when a is empty (to 2^31-1)

**Review by lrvideckis — CHANGES_REQUESTED** [review]

File: rangeDS/fenwickTree.cpp (hunk 2 above)
Comment (cameroncuster) [reply rangeDS/fenwickTree.cpp:5]: constructing a vector of size 0 will always RTE, I'm pretty sure...

File: rangeDS/fenwickTree.cpp (hunk 3 above)
Comment (cameroncuster) [reply rangeDS/fenwickTree.cpp]: Done!

**Pull request merged**